website/api/zkvm/developer-guide/profiling.md

```diff
@@ -1,192 +1,96 @@
-# let pprof_out = match std::env::var("RISC0_PPROF_OUT") {
-#     Ok(val) => Some(val),
-#     Err(_) => None,
-# };
-# let mut profiler = match pprof_out {
-#     Some(path) => Some(Profiler::new(&path, FIBONACCI_ELF).expect("profiler creation failed")),
-#     None => None,
-# };
-# let iterations = 1000;
-# let env = {
-#     let mut builder = ExecutorEnv::builder();
-#     if let Some(ref mut p) = profiler {
-#         builder.trace_callback(p.make_trace_callback());
-#     }
-#     builder
-#         .write_slice(&[iterations])
-#         .build()
-#         .expect("environment build failed")
-# };
-let exec = default_executor();
-exec.execute_elf(env, FIBONACCI_ELF).expect("execution failed");
-```
+[Sampling CPU profilers], as implemented by pprof and perf, provide a view of where your program is spending its time.
+It does so by recording the current call stack at a sampling interval.
+RISC Zero provides a "sampling" [^1] CPU profiler for guest execution.
 
-This will only [execute] the guest code, without generating a [receipt].
-
-5. Write out the profile
-
-```rust
-# use fibonacci_methods::{FIBONACCI_ELF, FIBONACCI_PATH};
-# use risc0_zkvm::{default_executor, ExecutorEnv, Profiler};
-# let pprof_out = match std::env::var("RISC0_PPROF_OUT") {
-#     Ok(val) => Some(val),
-#     Err(_) => None,
-# };
-# let mut profiler = match &pprof_out {
-#     Some(path) => Some(Profiler::new(&path, FIBONACCI_ELF).expect("profiler creation failed")),
-#     None => None,
-# };
-# let iterations = 1000;
-# let env = {
-#     let mut builder = ExecutorEnv::builder();
-#     if let Some(ref mut p) = profiler {
-#         builder.trace_callback(p.make_trace_callback());
-#     }
-#     builder
-#         .write_slice(&[iterations])
-#         .build()
-#         .expect("environment build failed")
-# };
-# let exec = default_executor();
-# exec.execute_elf(env, FIBONACCI_ELF).expect("execution failed");
-if let Some(ref mut p) = profiler {
-    p.finalize();
-    let report = p.encode_to_vec();
-    std::fs::write(pprof_out.as_ref().unwrap(), &report)
-        .expect("Unable to write profiling output");
-}
-```
+[pprof]: https://github.com/google/pprof
+[perf]: https://perf.wiki.kernel.org/index.php/Main_Page
+[Sampling CPU profilers]: https://nikhilism.com/post/2018/sampling-profiler-internals-introduction/
 
 ## Usage
 
 ### Step 1: Prerequisites
 
-First, install the [pprof] tool with
+First, [install Rust] if you don't already have it.
+Additionally, you will need to [install Go], which bundles with it the [pprof] tool.
+
+Next, install the `cargo-risczero` tool and install the toolchain with:
 
 ```bash
-go install github.com/google/pprof@latest
+cargo install cargo-binstall
+cargo binstall cargo-risczero
+cargo risczero install
 ```
 
 ### Step 2: Running
 
-Then, run the example with:
+Run the Fibonacci profiling example with:
 
 ```bash
-RISC0_PPROF_OUT=./profile.pb cargo run --release
+RISC0_PPROF_OUT=./profile.pprof cargo run
```

Comment:
I wonder why we're dropping `--release`

Reply:
It's not needed in the client-server scenario and `cargo run` is faster when not compiling in `release` mode. I can add it back though. I can definitely see the downside that if `prove` gets included, then building in debug mode would be an issue.

Reply:
Gotcha, yes linking can take a lot longer with `--release`. I'm fine with this, thanks.